src/python/pants/core/util_rules/environments.py

```diff
@@ -625,8 +626,38 @@ async def get_target_for_environment_name(
     return EnvironmentTarget(tgt)
 
 
+@rule_helper
+async def _maybe_add_docker_image_id(image_name: str, platform: Platform) -> str:
+    # If the image name appears to be just an image ID, just return it as-is.
+    if image_name.startswith("sha256:"):
+        return image_name
+
+    # If the image name contains an appended image ID, then use it.
+    if "@" in image_name:
+        image_name_part, _, maybe_image_id = image_name.rpartition("@")
+        if not maybe_image_id.startswith("sha256:"):
+            raise ValueError(
+                f"Docker image `{image_name}` contains an image ID component, but it does not appear to be a image SHA-256 hash."
```

Comment:
It'd be helpful to have a `description_of_origin`, which in this case should always be the `env_target.val.address`

> The Docker image `{image_name}` from the field `{DockerImageField.alias}` for the target {addr} contains ...

Reply:
Done.